core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,1207 @@
+          x0: 0.07250000000000001,
+          y0: 0.15333333333333335,
+          xLabel: 0.1625,
+          yLabel: 0.23333333333333334,
+          id: 'State1',
+          label: 'State1',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 1,
+          offset: 1,
+          reachable: true,
+          x0: 0.29750000000000004,
+          y0: 0.42000000000000004,
+          xLabel: 0.3875,
+          yLabel: 0.5,
+          id: 'State3',
+          label: 'State3',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.23 0.31333333333333335 Q 0.30557165934031566' +
+          ' 0.3408718290982754 0.32 0.42'
+      },
+      {
+        source: {
+          depth: 1,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.42000000000000004,
+          xLabel: 0.1625,
+          yLabel: 0.5,
+          id: 'State2',
+          label: 'State2',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 2,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.6866666666666666,
+          xLabel: 0.1625,
+          yLabel: 0.7666666666666666,
+          id: 'State4',
+          label: 'State4',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.1625 0.5800000000000001 Q 0.2025 0.6333333333333333' +
+          ' 0.1625 0.6866666666666665'
+      },
+      {
+        source: {
+          depth: 1,
+          offset: 1,
+          reachable: true,
+          x0: 0.29750000000000004,
+          y0: 0.42000000000000004,
+          xLabel: 0.3875,
+          yLabel: 0.5,
+          id: 'State3',
+          label: 'State3',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        target: {
+          depth: 2,
+          offset: 0,
+          reachable: true,
+          x0: 0.07250000000000001,
+          y0: 0.6866666666666666,
+          xLabel: 0.1625,
+          yLabel: 0.7666666666666666,
+          id: 'State4',
+          label: 'State4',
+          height: 0.16,
+          width: 0.18000000000000002,
+          reachableFromEnd: false
+        },
+        d: 'M0.32 0.5800000000000001 Q 0.30557165934031566' +
+          ' 0.6591281709017246 0.23000000000000004 0.6866666666666665'
+      }
+    ];
+
+    let returnedAugmentedLinks = sgls.getAugmentedLinks(nodeData, links2);
+    for (var i = 0; i < returnedAugmentedLinks.length; i++) {
+      expect(returnedAugmentedLinks[i].d).toBe(expectedAugmentedLinks[i].d);
+    }
+  });
+
+  it('should return undefined when source and target nodes overlap' +
+    ' while processing augmented links', () => {
```

Comment:
Perhaps leave a comment to explain why it is clear from the nodeData dict that some nodes are overlapping.

Also try to use the minimal test data that exposes that condition.

Reply:
Added comment and reduced data.